src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/SetAzureRmAnalysisServicesServer.cs

```diff
@@ -147,7 +158,12 @@ public override void ExecuteCmdlet()
                     ReadonlyReplicaCount = -1;
                 }
 
-                AnalysisServicesServer updatedServer = AnalysisServicesClient.CreateOrUpdateServer(ResourceGroupName, Name, location, Sku, Tag, Administrator, currentServer, BackupBlobContainerUri, ReadonlyReplicaCount, DefaultConnectionMode, setting);
+                if (DisassociateGateway.IsPresent)
+                {
+                    GatewayResourceId = "-";
```

Comment:
I would like to see this special value, namely "-" defined in a common file as a constant in some common codethat can be comsumed from the various places this is used.